pdu/UDH.go

```diff
@@ -113,9 +113,15 @@ type InfoElement struct {
 
 // NewIEConcatMessage  turn a new IE element for concat message info
 // IE.Data is populated at time of object creation
-func NewIEConcatMessage(totalParts, partNum, mref int) InfoElement {
-	return InfoElement{
-		ID:   data.UDH_CONCAT_MSG_8_BIT_REF,
-		Data: []byte{data.UDH_CONCAT_MSG_8_BIT_REF, 0x03, byte(mref), byte(totalParts), byte(partNum)},
+func NewIEConcatMessage(totalParts, sequence byte, reference uint16) InfoElement {
+	var element InfoElement
+	if reference > 0xFF {
+		element.ID = data.UDH_CONCAT_MSG_16_BIT_REF
+		element.Data = []byte{0, 0, totalParts, sequence}
```

Comment:
0, 0 is wrong. Please marshal BE “reference”

Reply:
`0, 0` is holder, on next line fill here.

```go
binary.BigEndian.PutUint16(element.Data[0:2], reference)
```

Reply:
Ah, I misread the next line of filtering. I am sorry about that